.github/workflows/update.yaml

```diff
@@ -0,0 +1,113 @@
+          
+          # Get latest stable Go versions
+          GO_STABLE_MAJOR_VERSIONS=$(curl -sSLf https://go.dev/dl/?mode=json | jq -r '[.[] | select(.stable == true) | .version | sub("^go"; "") | split(".")[:2] | join(".")] | unique | .[]')
+          if [ -z "${GO_STABLE_MAJOR_VERSIONS}" ]; then
+            echo "Failed to fetch stable Go versions"
+            exit 1
+          fi
+
+          # Check for differences between GO_MAJOR_VERSIONS and GO_STABLE_MAJOR_VERSIONS
+          DIFF=$(diff <(printf "%s\n" ${GO_MAJOR_VERSIONS}) <(printf "%s\n" ${GO_STABLE_MAJOR_VERSIONS}))
+          if [ -n "${DIFF}" ]; then
+            echo "⚠️ **Major Version Update Required**" >> ./update-body.txt
+            echo "Differences detected between specified and stable Go versions:" >> ./update-body.txt
+            echo "${DIFF}" >> ./update-body.txt
+            echo "" >> ./update-body.txt
+          fi
+          
+          # Function to update Dockerfile and TAG file
+          update_files() {
+            local go_major=$1
+            local codename=$2
+            local latest_version=$3
+            local dockerfile_path="golang-all/golang-${go_major}-${codename}/Dockerfile"
+            local tag_path="golang-all/golang-${go_major}-${codename}/TAG"
+
+            if [ ! -f "${dockerfile_path}" ] || [ ! -f "${tag_path}" ]; then
+              echo "Files not found for Go ${go_major} on Ubuntu ${codename}"
+              return
+            fi
+
+            local current_version=$(grep 'ARG GO_VERSION=' "${dockerfile_path}" | cut -d= -f2)
+
+            if [ "${current_version}" != "${latest_version}" ]; then
+              NEED_UPDATE=1
+              # Update Dockerfile
+              sed -i "s/ARG GO_VERSION=.*/ARG GO_VERSION=${latest_version}/" "${dockerfile_path}"
+              # Update TAG file
+              echo "${latest_version}_${codename}" > "${tag_path}"
+              
+              echo "- Update Go ${go_major} for Ubuntu ${ubuntu_version} (${codename}) from ${current_version} to ${latest_version}" >> ./update-body.txt
+            fi
+          }
+
+          # assignment of codename
+          for ubuntu_version in ${UBUNTU_VERSIONS}; do
+            if [ "${ubuntu_version}" = "22.04" ]; then
+              CODENAME=jammy
+            elif [ "${ubuntu_version}" = "24.04" ]; then
+              CODENAME=noble
+            else
+              echo "Unknown Ubuntu version: ${ubuntu_version}"
+              continue
+            fi
+            
+            for go_major in ${GO_MAJOR_VERSIONS}; do
+              # Find matching version from stable versions
+              LATEST_VERSION=$(echo "${GO_STABLE_MAJOR_VERSIONS}" | grep "^${go_major}" | head -n1)
+              if [ -z "${LATEST_VERSION}" ]; then
+                echo "No matching stable version found for Go ${go_major}"
+                continue
+              fi
+              
+              update_files "${go_major}" "${CODENAME}" "${LATEST_VERSION}"
```

Comment:
`LATEST_VERSION` does not include a minor revision number, but it is compared to `GO_VERSION` which includes a minor revision number. I think this is unintended behavior, please fix.
https://github.com/cybozu/neco-containers/blob/4b958bc1b3c19f46e70fd80ac4277d9b7c7ffd71/.github/workflows/update.yaml#L58-L60